### PR TITLE
revamp overview of ingestion

### PR DESCRIPTION
**Pull request by st1page**
## Related doc issue

Resolve

## Checklist

- [ ] I have checked the doc site preview, and the updated parts look good.
- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

File: docs/ingest/data-ingestion.md [hunk 1]
```diff
@@ -7,69 +7,79 @@ slug: /data-ingestion
...
+2. **Allow defining primary keys:** With the help of its internal storage, RisingWave can efficiently maintain primary key constraints. Users can define a primary key on a specific column of the table and define different behaviors for primary key conflicts with [ON CONFLICT clause](/sql/commands/sql-create-table.md#pk-conflict-behavior).
+3. **Ability to handle delete/update changes**: Based on the definition of primary keys, RisingWave can efficiently process upstream synchronized delete and update operations. For systems that synchronize delete/update operations from external systems, such as database's CDC, we **do not** allow creating source on it but require a table with connector.
 
-To ingest data from these sources, you need to create a source or table and specify the connector settings in the `CREATE SOURCE` or `CREATE TABLE` statements.
+At the same time, like regular tables, tables with connectors also accept DML statements and [CREATE SINK INTO TABLE](/sql/commands/sql-create-sink-into.md), which provides greater flexibility.
 
-## Insert data into tables (without connectors)
+## DML on tables
```
Comment (st1page) [inline docs/ingest/data-ingestion.md:83]: todo in future PR: move https://docs.risingwave.com/docs/next/deletes-and-updates/ here.

File: docs/ingest/data-ingestion.md [hunk 2]
```diff
@@ -86,6 +96,29 @@ INSERT INTO website_visits (timestamp, user_id, page_id, action) VALUES
...
+
+create table t1(
+  timestamp timestamp with time zone,
+  user_id varchar,
+  page_id varchar,
+  action varchar
+);
+
+insert into t1 select * from source_iceberg_t1;
+```
```
Comment (st1page) [inline docs/ingest/data-ingestion.md:128]: todo in future PR: add guide for merge bulk data and later streaming ingestion

File: docs/ingest/data-ingestion.md [hunk 3]
```diff
@@ -7,69 +7,79 @@ slug: /data-ingestion
...
+  SELECT * FROM source_name
+  WHERE _rw_kafka_timestamp > now() - interval '10 minute';
+  ```
 
-Regardless of whether data is persisted in RisingWave, you can create materialized views to transform or analyze them.
-  </div>
-</details>
+For specific source types, their support for Streaming Ingestion and ad-hoc ingestion varies. Please refer to the [doc](/docs/current/sources) for the specific sources.
 
+### Table with connector
```
Comment (st1page) [inline docs/ingest/data-ingestion.md]: todo in future PR: I used to think to add some information about what happen when create a streaming query on the table. But I soon realized it should be a general explaination for users. It should not only for the table with connector but for all MV on MV and MV on table

File: docs/ingest/data-ingestion.md [hunk 4]
```diff
@@ -7,69 +7,79 @@ slug: /data-ingestion
...
 
-### Supported source types
+For sources that support streaming ingestion, RisingWave supports the direct creation of tables on them, which called "Table with connector"([`CREATE TABLE`](/sql/commands/sql-create-table.md))
```
Comment (fuyufjh) [inline docs/ingest/data-ingestion.md]: No need to introduce a term for it.

```suggestion
For sources that support streaming ingestion, RisingWave supports the direct creation of tables on them.
```

File: docs/ingest/data-ingestion.md [hunk 5]
```diff
@@ -7,69 +7,79 @@ slug: /data-ingestion
...
-<br/>
-
-<div>As shown above, a key distinction between the two is that a table persists the ingested raw data, whereas a source does not. For example, let's consider the upstream input of 5 records: `AA`, `BB`, `CC`, `DD`, and `EE`. If a table is used, these 5 records will be persisted in RisingWave. However, if a source is used, these records will not be persisted. </div>
-<br/>
-<div>One advantage of using a table is that you can perform ad-hoc queries against the ingested raw data. </div>
-<br/>
-
-<div>Another advantage of using a table is the ability to consume data changes. If the upstream system deletes or updates a record, this operation will be consumed by RisingWave, thereby modifying the results of the stream computation. On the other hand, a source only supports appending records and cannot handle data changes. Besides, to allow a table to accept data changes, a primary key must be specified on the table.</div>
+RisingWave supports a variety of data ingestion methods. To know the difference between stream processing and ad-hoc query, please refer to [Ad hoc (on read) vs. Streaming (on write)](/transform/overview.md#ad-hoc-on-read-vs-streaming-on-write)
```
Comment (fuyufjh) [inline docs/ingest/data-ingestion.md]:
```suggestion
RisingWave supports a variety of data ingestion methods. To know the difference between stream processing and ad-hoc query, please refer to [Ad-hoc (on read) vs. Streaming (on write)](/transform/overview.md#ad-hoc-on-read-vs-streaming-on-write)
```

File: docs/ingest/data-ingestion.md [hunk 6]
```diff
@@ -86,6 +96,29 @@ INSERT INTO website_visits (timestamp, user_id, page_id, action) VALUES
   ('2023-06-13T10:04:00Z', 'user5', 'page2', 'view');
 ```
 
+### Use `insert ... select ...` do bulk ingestion
```
Comment (fuyufjh) [inline docs/ingest/data-ingestion.md]:
```suggestion
### Use `INSERT SELECT` to do bulk ingestion
```

File: docs/ingest/data-ingestion.md [hunk 7]
```diff
@@ -7,69 +7,79 @@ slug: /data-ingestion
...
-<div>Apart from the above differences, here are a few points worth noting about a table:</div>
-<br/>
-<div></div>
+After creating a source, no actual data ingestion will occur. Data ingestion happens when a query that references the source is submitted. 
 
-- With a `CREATE TABLE` statement, the corresponding table will be immediately created and populated with data.
-- When a materialized view is defined based on the existing table, RisingWave will start reading data from the table and perform streaming computation.
-- RisingWave's batch processing engine supports direct batch reading of the table. Users can issue ad-hoc queries against the data within the table.
+- `CREATE MATERIALIZED VIEW` or `CREATE SINK` statement will generate an **streaming ingestion** jobs. 
+The following statement will continuously ingest content from the Kafka topic and maintain it in the materialized view `mv`.
```
Comment (fuyufjh) [inline docs/ingest/data-ingestion.md]:
```suggestion
The following statement will continuously import data from the Kafka topic and store it in the materialized view `mv`.
```

File: docs/ingest/data-ingestion.md [hunk 8]
```diff
@@ -7,69 +7,79 @@ slug: /data-ingestion
...
-And here are the points worth noting about a source:
+  ```SQL
+  CREATE MATERIALIZED VIEW mv AS
+  SELECT *
+  FROM kafka_source;
+  ```
 
-- With a `CREATE SOURCE` statement, no physical objects are created, and data is not immediately read from the source.
-- Data from the source is only read when a user creates materialized views or sinks on that source.
+- Also, directly query can be processed on the source and an **ad-hoc ingestion** will happened during the query's processing (more information in [directly query kafka](/ingest/ingest-from-kafka.md#query-kafka-timestamp))
```
Comment (fuyufjh) [inline docs/ingest/data-ingestion.md]:
```suggestion
- Also, queries can be executed directly on the source, and **ad-hoc ingestion** will happened during the query's processing (more information in [directly query kafka](/ingest/ingest-from-kafka.md#query-kafka-timestamp))
```
```suggestion
- Also, directly query can be processed on the source and an **ad-hoc ingestion** will happened during the query's processing (more information in [directly query kafka](/ingest/ingest-from-kafka.md#query-kafka-timestamp))
```

**Review by fuyufjh — COMMENTED** [review]
The structure generally LGTM, but the English writing may need to be polished

**Review by ShanlanLi — APPROVED** [review]
Polished the language and format, LGTM

**Review by xxchan — COMMENTED** [review]
I think we should link back to this page in `CREATE SOURCE` 🥸

**Comment by hengm3467** [comment]
> I think we should link back to this page in `CREATE SOURCE` 🥸

Agree! :)

File: docs/ingest/data-ingestion.md [hunk 9]
```diff
@@ -7,69 +7,85 @@ slug: /data-ingestion
...
 
-A source is a connection to an external system that RisingWave can read data from. RisingWave supports these types of sources:
+3. **Ability to handle delete/update changes**: Based on the definition of primary keys, RisingWave can efficiently process upstream synchronized delete and update operations. For systems that synchronize delete/update operations from external systems, such as database's CDC, we **do not** allow creating a source on it but require a table with connectors.
 
-- Event streaming systems such as Apache Kafka, Apache Pulsar, AWS Kinesis
-- Change data capture streams from databases such as MySQL, PostgreSQL, and MongoDB
-- Storage systems such as AWS S3 or S3-compatible systems
+At the same time, like regular tables, tables with connectors also accept DML statements and [CREATE SINK INTO TABLE](/sql/commands/sql-create-sink-into.md), which provides greater flexibility.
```
Comment (xxchan) [inline docs/ingest/data-ingestion.md:81]: Why not make this bullet point 4? I thought this point was missed until read carefully 🤪

Comment (xxchan) [reply docs/ingest/data-ingestion.md:81]: Oh, there's a whole section below for DML

**Pull request merged**